src/Configuration/GroupCombination.php

```diff
@@ -7,41 +7,32 @@
 class GroupCombination
 {
     /**
-     * @var ClassToGenerate
+     * @param list<string>      $groups
+     * @param list<string>|null $versions
      */
-    private $containingClass;
-
-    /**
-     * One combination of groups to generate.
-     *
-     * An empty array means to generate with no groups.
-     * e.g. ['api', 'details'].
-     *
-     * @var string[]
-     */
-    private $groups;
-
-    /**
-     * List of versions to generate.
-     *
-     * An empty string '' means to generate without a version.
-     * e.g. ['', '2', '3']
-     *
-     * If not specified, this falls back to the class default.
-     * If the array is not null, it must have a length > 0.
-     *
-     * @var string[]|null
-     */
-    private $versions;
-
-    public function __construct(ClassToGenerate $containingClass, array $groups, array $versions = null)
-    {
-        $this->containingClass = $containingClass;
-        $this->groups = $groups;
+    public function __construct(
+        private ClassToGenerate $containingClass,
+        /**
+         * One combination of groups to generate.
+         * An empty array means to generate with no groups.
+         *  e.g. ['api', 'details'].
+         */
```

Comment:
here we have the comment on the promoted argument and not as `@param` on the constructor method

Reply:
this constructor is for "all promoted" arguments, the first one is mixing promoted and non-promoted ones

I also would like to use the same pattern, but this conflicts with php-cs-fixer :(